platform/android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -154,24 +157,26 @@ open class PolyOut(
                     )
             }
         }
+
+        val zipId = destUrl ?: UUID.randomUUID().toString()
+
         supervisorScope {
             this.async(Dispatchers.IO) {
                 contentResolver?.openInputStream(uri).use { inputStream ->
                     if (inputStream == null) {
                         throw Error("File import error")
                     }
-                    val newId = UUID.randomUUID().toString()
                     val fs = Preferences.getFileSystem(context).toMutableMap()
-                    fs[fsPrefix + newId] = fileName
+                    fs[fsPrefix + zipId] = fileName
```

Comment:
Why this is different from iOS where a list of filenames is stored for the same path?

Reply:
I missed that. Good catch.